### PR TITLE
Replace some use of string.CompareOrdinal with string ==

### LINKED ISSUE #34514: Prefer string.Equals over string.Compare(string, string) == 0 or != 0
As a carryover from native code habits, I see folks use `string.Compare(a, b) == 0` or `!= 0` a *lot* where `string.Equals(a, b)` is more semantically meaningful.

But *more importantly*, Compare is for *sorting* strings whereas Equals is for determining equality. In particular cultures two strings may have *sort* equivalence without being *equal*. So in some cases it may even make a security difference (when StringComparison isn't specified). 

**Category**: Performance

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/runtime/issues/34514#issuecomment-1438713985

### CODE REVIEW

File: src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs [hunk 1]
```diff
@@ -501,7 +501,7 @@ bool IEqualityComparer<string>.Equals(string? extensionA, string? extensionB)
                 //with the rules for comparing/normalizing partnames.
                 //Refer to PackUriHelper.ValidatedPartUri.GetNormalizedPartUri method.
                 //Currently normalization just involves upper-casing ASCII and hence the simplification.
-                return (string.CompareOrdinal(extensionA.ToUpperInvariant(), extensionB.ToUpperInvariant()) == 0);
+                return extensionA.ToUpperInvariant() == extensionB.ToUpperInvariant();
```
Comment (EgorBo) [inline src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs:504]: nit: could be equals ignore case since the comment mentions ASCII?

**Review by EgorBo — APPROVED** [review]

File: src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs:504]: Yeah, but there's that scary comment about keeping all logic here in sync, so... yeah... not going to risk anything here. :)

File: src/libraries/System.Private.CoreLib/src/System/Globalization/CalendarData.Icu.cs [hunk 2]
```diff
@@ -435,7 +435,7 @@ private static unsafe void EnumCalendarInfoCallback(char* calendarStringPtr, Int
                 {
                     foreach (string existingResult in callbackContext->Results)
                     {
-                        if (string.CompareOrdinal(calendarStringSpan, existingResult) == 0)
+                        if (calendarStringSpan.SequenceEquals(existingResult))
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/CalendarData.Icu.cs:438]:
```suggestion
                        if (calendarStringSpan.SequenceEqual(existingResult))
```

File: src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs (hunk 1 above)
Comment (drieseng) [inline src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs:504]: Use `extensionA.Equals(extensionB, StringComparison.InvariantCultureIgnoreCase)` instead?

Comment (stephentoub) [reply src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipPackage.cs:504]: Maybe, but as in https://github.com/dotnet/runtime/pull/82438#discussion_r1113349780, I've avoided making any such changes so at to not accidentally break something subtle

**Pull request merged**